splice_ck/src/main/java/com/splicemachine/ck/command/RootCommand.java

```diff
@@ -14,19 +14,54 @@
 
 package com.splicemachine.ck.command;
 
+import com.splicemachine.ck.ConnectionCache;
+import com.splicemachine.ck.Utils;
 import org.apache.log4j.Level;
 import org.apache.log4j.Logger;
 import picocli.CommandLine;
 import picocli.CommandLine.Command;
 
-@Command(mixinStandardHelpOptions = true, name = "spliceck", description = "SpliceMachine check command suite", descriptionHeading = "Description:%n",
+import java.util.Arrays;
+import java.util.Scanner;
+
+@Command(mixinStandardHelpOptions = true, name = "spliceck", description = "SpliceMachine check command suite",
+        descriptionHeading = "Description:%n",
         optionListHeading = "Options:%n", subcommands = {TListCommand.class,
         TColsCommand.class, RegionOfCommand.class, TableOfCommand.class, RGetCommand.class, RPutCommand.class,
         TxListCommand.class, SListCommand.class})
 class RootCommand {
     public static void main(String... args) {
         Logger.getRootLogger().setLevel(Level.OFF);
-        int exitCode = new CommandLine(new RootCommand()).setExecutionStrategy(new CommandLine.RunLast()).execute(args);
+        ConnectionCache c = new ConnectionCache();
+
+        int exitCode = 0;
+        if(!args[0].equals("interactive")) {
+            exitCode = new CommandLine(new RootCommand()).setExecutionStrategy(new CommandLine.RunLast()).execute(args);
+        }
+        else {
+            Scanner command = new Scanner(System.in);
+            CommandLine cl = new CommandLine(new RootCommand()).setExecutionStrategy(new CommandLine.RunLast());
+            while (true) {
+                System.out.print("spliceck> ");
+                String s = command.nextLine();
+                if (s.equals("exit")) break;
+                if (s.trim().length() == 0) continue;
+                String[] args2 = s.split(" ");
+                exitCode = cl.execute(args2);
+                if( exitCode == 0 )
+                    System.out.println(Utils.Colored.green( "return code " + exitCode + "\n") );
+                else
+                    System.out.println(Utils.Colored.red( "return code " + exitCode + "\n") );
+
+            }
+            command.close();
+        }
+
+        try {
+            c.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
```

Comment:
I think this is redundant, `ConnectionWrapper` is `AutoCloseable`, the static object will be destroyed when the process finishes closing it resource.

Reply:
i wanted to make sure here we're closing the hbase connection as soon as possible